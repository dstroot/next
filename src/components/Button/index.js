import React from 'react';
import { Link } from '@reach/router';

const Button = props => {
  return (
    <Link to={props.path}>
      <a href={props.href} target="_blank">
        <button className="btn btn-primary rounded-pill nxt_body-small px-4 py-2">
          {props.buttonText}
        </button>
      </a>
    </Link>
  );
};

export default Button;
