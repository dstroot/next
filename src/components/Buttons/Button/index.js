import React from 'react';
import '.././styles.scss';
import { Link } from '@reach/router';

const Button = props => {
  return (
    <Link to={props.path}>
      <button className="btn btn-primary nxt_rounded-pill nxt_body-small px-4 py-2">
        {props.buttonText}
      </button>
    </Link>
  );
};

export default Button;
