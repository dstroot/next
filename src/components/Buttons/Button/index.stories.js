import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from '../Button';

storiesOf('NEXT|sections|Button', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show Button', () => <Button />);
