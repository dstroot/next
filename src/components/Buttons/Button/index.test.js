import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '.';

describe('Button', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Button path="/" />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<Button path="/" />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
