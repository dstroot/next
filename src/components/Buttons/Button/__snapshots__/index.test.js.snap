// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should render 1`] = `
<a
  aria-current="page"
  href="/"
  onClick={[Function]}
>
  <button
    className="btn btn-primary nxt_rounded-pill nxt_body-small px-4 py-2"
  />
</a>
`;
